lib/backend_solver.ml

```diff
@@ -0,0 +1,65 @@
+open Lwt.Infix
+
+let ( >>!= ) = Lwt_result.bind
+
+type t =
+  [ `Remote of Current_ocluster.Connection.t
+  | `Local of Ocaml_ci_api.Solver.t Lwt.t ]
```

Comment:
Since these constructors aren't used elsewhere they can just be regular variants.